go/vt/dbconfigs/dbconfigs.go

```diff
@@ -246,8 +246,12 @@ func Init(defaultSocketFile string) (*DBConfigs, error) {
 			uc.param.UnixSocket = defaultSocketFile
 		}
 
-		uc.param.Charset = baseConfig.Charset
-		uc.param.Flags = baseConfig.Flags
+		if baseConfig.Charset != "" {
```

Comment:
it seems like the default `baseConfig.Charset` is `utf8`: https://github.com/planetscale/vitess/blob/65138f200afdf5d427e415dde9e22be5d5340f7d/go/vt/dbconfigs/dbconfigs.go#L97

Won't this only work if people override the `baseConfig.Charset` to be the empty string? I'd imagine most people who are using userconfig charsets are not overriding `baseConfig.Charset` to be the empty string?